x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/amazonbedrock/completion/AmazonBedrockCompletionTaskSettings.java

```diff
@@ -78,27 +78,27 @@ public static AmazonBedrockChatCompletionTaskSettings fromMap(Map<String, Object
             throw validationException;
         }
 
-        return new AmazonBedrockChatCompletionTaskSettings(temperature, topP, topK, maxNewTokens);
+        return new AmazonBedrockCompletionTaskSettings(temperature, topP, topK, maxNewTokens);
     }
 
-    public static AmazonBedrockChatCompletionTaskSettings of(
-        AmazonBedrockChatCompletionTaskSettings originalSettings,
-        AmazonBedrockChatCompletionRequestTaskSettings requestSettings
+    public static AmazonBedrockCompletionTaskSettings of(
+        AmazonBedrockCompletionTaskSettings originalSettings,
+        AmazonBedrockCompletionRequestTaskSettings requestSettings
     ) {
         var temperature = requestSettings.temperature() == null ? originalSettings.temperature() : requestSettings.temperature();
         var topP = requestSettings.topP() == null ? originalSettings.topP() : requestSettings.topP();
         var topK = requestSettings.topK() == null ? originalSettings.topK() : requestSettings.topK();
         var maxNewTokens = requestSettings.maxNewTokens() == null ? originalSettings.maxNewTokens() : requestSettings.maxNewTokens();
```

Comment:
If `requestSettings` is empty, we can just return `originalSettings`. Similarly, if the values to use are equal to the values in the original settings, we don't need to create a new settings object and can just return the original ones.